tests/framework/matcher/status.go

```diff
@@ -82,3 +84,26 @@ func HaveStateChangeRequests() gomegatypes.GomegaMatcher {
 		}),
 	}))
 }
+
+func InterfaceByName(interfaceName string) func(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstanceNetworkInterface {
+	return func(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstanceNetworkInterface {
+		for _, networkInterface := range vmi.Status.Interfaces {
+			if networkInterface.Name == interfaceName {
+				return &networkInterface
+			}
+		}
+		return nil
+	}
+}
+
+func InterfaceIPs(networkInterface *v1.VirtualMachineInstanceNetworkInterface) []string {
+	GinkgoHelper()
+	Expect(networkInterface).NotTo(BeNil(), "missing network interface")
```

Comment:
nit: keep in mind I'm being pedantic here, and only doing so because the test / new matcher below look really good. 

I really dislike this expect in the helper; maybe it would make sense to have `InterfaceByName` return a `v1.VirtualMachineInstanceNetworkInterface` (empty one) if it isn't available. Thus, you can avoid both the ginkgo helper and the expect not to be nil here.

IMHO that makes sense because afaict the sole purpose of `InterfaceByName` is to used here.

Reply:
I am going to add some more composition and put this "Expect" as part of the composition chain.

Reply:
I guess that's an acceptable compromise.